src/blocks/affiliate/shared/helpers.js

```diff
@@ -1,5 +1,36 @@
-export const refreshAffiliateWidgets = () => {
-  window.organic ||= {};
-  window.organic.cmd ||= [];
-  window.organic.cmd.push(({ affiliate }) => affiliate?.processPage());
+import { select, subscribe } from '@wordpress/data';
+
+// See https://gist.github.com/KevinBatdorf/fca19e1f3b749b5c57db8158f4850eff
+async function whenEditorIsReady() {
+  return new Promise((resolve) => {
+    const unsubscribe = subscribe(() => {
+      // This will trigger after the initial render blocking, before the window load event
+      // This seems currently more reliable than using __unstableIsEditorReady
+      if (select('core/editor').isCleanNewPost() || select('core/block-editor').getBlockCount() > 0) {
+        unsubscribe();
+        resolve();
+      }
+    });
+  });
+}
+
+const refreshAffiliateWidgets = () => {
+  setTimeout(() => {
+    window.organic ||= {};
+    window.organic.cmd ||= [];
+    window.organic.cmd.push(({ affiliate }) => affiliate?.processPage());
+  }, 0);
+};
+
+export const refreshAffiliateWidgetsOnEdit = () => {
+  refreshAffiliateWidgets();
+};
+
+export const refreshAffiliateWidgetsOnSave = () => {
+  whenEditorIsReady().then(() => {
+    if (!window.wp_organic_affiliate_processed) {
```

Comment:
I can add a comment, something like, "Prevent multiple calls to pageProcess."

Reply:
(Since every widget on the admin page will call this.)

Reply:
Adding a comment sounds like a good idea; another suggestion is to use `window.__wpOrganicAffiliateProcessed` flag for consistency